browser-extension/src/background.js

```diff
@@ -192,42 +186,36 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     };
 
     if (request.message === 'getTournesolRecommendations') {
-      // Compute the number of videos to load in each category
-      const recentVideoToLoad = Math.round(
-        request.videosNumber *
-          oversamplingRatioForRecentVideos *
-          recentVideoProportion
-      );
-      const oldVideoToLoad = Math.round(
-        request.videosNumber *
-          oversamplingRatioForOldVideos *
-          (1 - recentVideoProportion)
-      );
-      const recentAdditionalVideoToLoad = Math.round(
-        request.additionalVideosNumber *
-          oversamplingRatioForRecentVideos *
-          recentVideoProportionForAdditionalVideos
+      const api_path = `polls/${poll_name}/recommendations/random/`;
+
+      const nbrPerRow = request.videosNumber;
+      const extraNbr = request.additionalVideosNumber;
+
+      const recentToLoadRow1 = Math.round(nbrPerRow * RECENT_VIDEOS_RATIO);
+      const oldToLoadRow1 = Math.round(nbrPerRow * (1 - RECENT_VIDEOS_RATIO));
+
+      const recentToLoadExtra = Math.round(
+        extraNbr * RECENT_VIDEOS_EXTRA_RATIO
       );
-      const oldAdditionalVideoToLoad = Math.round(
-        request.additionalVideosNumber *
-          oversamplingRatioForOldVideos *
-          (1 - recentVideoProportionForAdditionalVideos)
+      const oldToLoadExtra = Math.round(
+        extraNbr * (1 - RECENT_VIDEOS_EXTRA_RATIO)
       );
 
       const process = async () => {
         const threeWeeksAgo = getDateThreeWeeksAgo();
         const recommendationsLangs =
           await getRecommendationsLanguagesAuthenticated();
 
-        // Only one request for both videos and additional videos
         const recentParams = new URLSearchParams([
           ['date_gte', threeWeeksAgo],
-          ['limit', recentVideoToLoad + recentAdditionalVideoToLoad],
+          ['limit', (recentToLoadRow1 + recentToLoadExtra) * BUNDLE_SIZE_F],
+          ['random', request.queryParamRandom],
```

Comment:
To my understanding this parameter could have any name and is just helpful to invalidate the cache?

maybe we could name it `cache_avoidance_param` ?


It looks a bit like a `random_seed` but that's not how it actually works either.

Reply:
We had this conversation with @amatissart .

`seed` / `random_seed` could have worked, but it's not really a seed, and using the same parameter won't necessarily return the same results after the cache has expired.

`cache_avoidance_param` could work, it's still a bit confusing to me. For instance calling several times in a row `cache_avoidance_param=true` will return the same cached results, even if the name suggests that it won't.

What do you think about `bundle`? `bundle=1` means we want the first cached video bundle identified by the integer 1.

Reply:
I like bundle the best :)

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1876/commits/4c1cb676aeca5578c0066e8cac502d78058f761a